### PR TITLE
tools: let environment-setup.sh skip unnecessary steps

### PR DESCRIPTION
**Pull request by clobrano**
Current environment-setup.sh implementation fails in some points when
the lxc container exists and it is configured - even only partially.

Added some more checks to skip already performed configuration steps.

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
    No, but the change does not affect source code
- [ ] Have you successfully run `./runtests.sh tests/unit`?
    No, but the change does not affect source code

-----

### CODE REVIEW

File: tools/environment-setup.sh [hunk 1]
```diff
@@ -14,9 +14,15 @@ if ! grep -q '^name: snapcraft$' snap/snapcraft.yaml; then
 fi
 
 # Create the container
-lxc init ubuntu:16.04 snapcraft-dev
-lxc config set snapcraft-dev raw.idmap "both $UID 1000"
-lxc start snapcraft-dev
+if ! lxc list | grep -q "snapcraft-dev"; then
```
Comment (cjp256) [inline tools/environment-setup.sh]: suggested improvement to improve accuracy:

`if ! lxc info snapcraft-dev >/dev/null 2>&1`

File: tools/environment-setup.sh [hunk 2]
```diff
@@ -14,9 +14,15 @@ if ! grep -q '^name: snapcraft$' snap/snapcraft.yaml; then
...
+    lxc init ubuntu:16.04 snapcraft-dev
+fi
+if ! lxc config get snapcraft-dev raw.idmap | grep -q "both $UID 1000"; then
+    lxc config set snapcraft-dev raw.idmap "both $UID 1000"
+fi
+if ! lxc list | grep "snapcraft-dev" | grep -q "RUNNING"; then
```
Comment (cjp256) [inline tools/environment-setup.sh]: suggested improvement to improve accuracy:

`if ! lxc info snapcraft-dev | grep -q "Status: Running"; then`

File: tools/environment-setup.sh (hunk 1 above)
Comment (clobrano) [reply tools/environment-setup.sh]: Thanks! I will update the PR

Comment (cjp256) [reply tools/environment-setup.sh]: Thank you for the updates!

**Comment by cjp256** [comment]
Thanks for the updates! LGTM

**Pull request merged**